core/types/receipt.go

```diff
@@ -612,26 +614,8 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, blockHash common.Has
 		logIndex += uint(len(rs[i].Logs))
 	}
 
-	if config.Optimism != nil && len(txs) >= 2 && config.IsBedrock(new(big.Int).SetUint64(blockNumber)) {
-		gasParams, err := extractL1GasParams(config, blockTime, txs[0].Data())
-		if err != nil {
-			return err
-		}
-		for i := 0; i < len(rs); i++ {
-			if txs[i].IsDepositTx() {
-				continue
-			}
-			rs[i].L1GasPrice = gasParams.l1BaseFee
-			rs[i].L1BlobBaseFee = gasParams.l1BlobBaseFee
-			rs[i].L1Fee, rs[i].L1GasUsed = gasParams.costFunc(txs[i].RollupCostData())
-			rs[i].FeeScalar = gasParams.feeScalar
-			rs[i].L1BaseFeeScalar = u32ptrTou64ptr(gasParams.l1BaseFeeScalar)
-			rs[i].L1BlobBaseFeeScalar = u32ptrTou64ptr(gasParams.l1BlobBaseFeeScalar)
-			if gasParams.operatorFeeScalar != nil && gasParams.operatorFeeConstant != nil && (*gasParams.operatorFeeScalar != 0 || *gasParams.operatorFeeConstant != 0) {
-				rs[i].OperatorFeeScalar = u32ptrTou64ptr(gasParams.operatorFeeScalar)
-				rs[i].OperatorFeeConstant = gasParams.operatorFeeConstant
-			}
-		}
+	if config.IsOptimismBedrock(new(big.Int).SetUint64(blockNumber)) && len(txs) >= 2 {
```

Comment:
Condition changed: we still need the `config.Optimism != nil`, correct? Suggesting we leave the condition unchanged since it doesn't seem to have an impact on this PR. 
```suggestion
	if config.Optimism != nil && len(txs) >= 2 && config.IsOptimismBedrock(new(big.Int).SetUint64(blockNumber)) {
```

Reply:
I think the condition is equivalent because of this: 

https://github.com/ethereum-optimism/op-geth/blob/96738d21616cd7c37101fd49234e361f70e7b64c/params/config.go#L907-L910

Reply:
Correct, using `config.Optimism != nil` is an outdated bad way to check for Optimism forks. In fact, we also think about removing all the `IsOptimism...` checks because any chain config with at least bedrock enabled must have a non-nil `Optimism` field.